src/npc.cpp

```diff
@@ -310,22 +311,22 @@ void Npc::onCreatureMove(Creature* creature, const Tile* newTile, const Position
 	}
 }
 
-void Npc::onCreatureSay(Creature* creature, SpeakClasses type, const std::string& text)
+void Npc::onCreatureSay(std::shared_ptr<Creature> creature, SpeakClasses type, const std::string& text)
 {
-	if (creature == this) {
+	if (creature.get() == this) {
 		return;
 	}
 
 	// only players for script events
-	Player* player = creature->getPlayer();
+	auto player = creature->getPlayer();
```

Comment:
Could this be `const auto`
```suggestion
	const auto player = creature->getPlayer();
```